### PR TITLE
refactor(themes): make map processing lazy, add variable existence check

### PR DESCRIPTION
**Pull request by nnixaa**

### CODE REVIEW

File: src/framework/theme/styles/core/theming/_get-value.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    // there is no way to check if variable exists as current execution context is outside of particular theme
+    // because we process css in this mode only once! (and not for each theme)
+    $value: var(--#{$key});
+  }
+
+  // in a preprocess mode (nb-install-global call) get ready value from $nb-processed-theme variable
+  @if ($nb-theme-process-mode == 'pre-process') {
+    $value: map-get($nb-processed-theme, $key);
+
+    // otherwise lazily search for variable value
```
Comment (yggg) [inline src/framework/theme/styles/core/theming/_get-value.scss]: Should be one line below

File: src/playground/styles/themes.scss [hunk 2]
```diff
@@ -1,15 +1,4 @@
...
-), default, default);
-
-$nb-themes: nb-register-theme((
-
-), cosmic, cosmic);
-
-$nb-themes: nb-register-theme((
-
-), corporate, corporate);
+$nb-enable-css-custom-properties: false;
```
[the remark this answers is not in the document]
Comment (nnixaa) [reply src/playground/styles/themes.scss]: It's just for a reference here, for easier access in the playground, to be able to change it quickly and check in css props mode.

File: src/framework/theme/styles/core/theming/_get-value.scss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+  @return $processed-theme;
+}
+
+@function nb-theme($key) {
+
+  $value: ();
+
+  // in case of css custom properties - just returns var(--var-name) - the rest is a browser job
+  @if ($nb-theme-process-mode == null) {
```
[the remark this answers is not in the document]
Comment (nnixaa) [reply src/framework/theme/styles/core/theming/_get-value.scss]: I believe having a common `mode` thing that controls the flow is more straightforward

File: src/framework/theme/components/input/_input.directive.theme.scss [hunk 4]
```diff
@@ -16,11 +16,7 @@
     border-style: nb-theme(input-border-style);
     border-width: nb-theme(input-border-width);
     color: nb-theme(input-text-color);
-    font-size: nb-theme(input-text-font-size);
```
[the remark this answers is not in the document]
Comment (nnixaa) [reply src/framework/theme/components/input/_input.directive.theme.scss]: yes, later on for a particular size

**Pull request merged**